lib/membership_row.rb

```diff
@@ -0,0 +1,37 @@
+require 'scraped'
+
+class MembershipRow < Scraped::HTML
+  TERM_IDS_MAP = {
+    'Legislatura' => '0',
+    'I' => '1',
+    'II' => '2',
+    'III' => '3',
+    'IV' => '4',
+    'V' => '5',
+    'VI' => '6',
+    'VII' => '7',
+    'VIII' => '8',
+    'IX' => '9',
+    'X' => '10',
+    'XI' => '11',
+    'XII' => '12'
+  }.freeze
+
+  field :term do
+    TERM_IDS_MAP[noko.at_css('.principal').text.split.first]
+  end
+
+  field :constituency do
+    constituency_and_faction_line.rpartition('(').first.tidy
+  end
+
+  field :faction do
+    constituency_and_faction_line.rpartition('(').last.split(')').first.tidy
```

Comment:
I don't know what this line looks like (a comment would be handy), but immediately calling `.first` (or `.last`) on `.rpartition` is a bit of a warning sign that that's probably not the best approach. If you had a separate method that was splitting the line up, which `:constituency` and `:faction` each used to get the part they cared about, that might make more sense. But without knowing what this line can look like, it's hard to know what the best approach is likely to be. In general, though, splitting on '(' and then again on ')' is likely to not really be a great approach, unless you're very deliberately needing to cope with odd layouts, potentially with oddly nested brackets (in which case you need more explanation.) This is one where a regex is generally going to be a lot clearer (especially when moved to a method that gets reused across both places that have to parse this line)